### DISCUSSION TITLE
MissingMethodException returned from remote dataportal for NameValueListBase class

### DISCUSSION THREAD

**Discussion by johnlabue** [MarimerLLC/csla#3082] [Questions]
Hi.  I am attempting to move my solution from CSLA 5.4 to 6.1.  I have a class library project containing many different classes which inherit from the various CSLA base classes (BusinessBase, ReadonlyBase, ReadonlyListBase, BusinessListBase).  All of these types have migrated successfully, however I have one class that inherits from NameValueListBase that is giving me a problem.  When I configure my solution to use a local dataportal, this NameValueListBase class works fine.  But when I use a remote dataportal (hosted by an ASP.NET Core 6 endpoint), the following System.MissingMethodException gets thrown back to the calling client when I attempt to instantiate an instance:

System.MissingMethodException: 'Constructor on type 'Csla.NameValueListBase`2+NameValuePair[[Csla6.Lib.CardTypeId, Csla6.Lib, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[System.String, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]' not found.'

Here is the signature of this class and corresponding enum, which are located in class library project "Csla6.Lib":

```
    [Serializable()]
    public enum CardTypeId : int
    {
        Password = 1,
        CreditCard = 2,
        Computer = 3,
        Miscellaneous = 4
    }
```

```
    [Serializable()]
    public class CardTypeList : NameValueListBase<CardTypeId, string>
```

Here is the client-code used to instantiate:

```
	var portal = _dataPortalFactory.GetPortal<CardTypeList>();
	CardTypeList cardTypeList = portal.Fetch();
```

When calling this portal.Fetch() method, I have been able to debug and successfully step into the CardTypeList.Fetch method, so I know this inbound call works.  It is only when the BO finishes executing this method on the remote dataportal and attempts to return back to the client that the above exception is thrown.

I first encountered this problem when using a Blazor WebAssembly project as my client app.  Assuming this was some issue with Blazor, I then created a simple console client application, pointed at the same remote dataportal endpoint, but this test client app also experienced the same exception.   

Any insight into this error would be greatly appreciated.

Thanks,

John

**Comment by rockfordlhotka** [comment]
I confirmed that this fails, so elevated to an issue: https://github.com/MarimerLLC/csla/issues/3083
